src/components/Banner/Banner.mdx

```diff
@@ -0,0 +1,184 @@
+import { ArgTypes, Canvas, Description, Meta, Source } from "@storybook/blocks";
+
+import * as BannerStories from "./Banner.stories";
+import Link from "../Link/Link";
+import ComponentChangelogTable from "../../utils/ComponentChangelogTable";
+import { changelogData } from "./bannerChangelogData";
+
+<Meta of={BannerStories} />
+
+# Banner
+
+| Component Version | DS Version   |
+| ----------------- | ------------ |
+| Added             | `0.23.2`     |
+| Latest            | `Prerelease` |
+
+## Table of Contents
+
+- {<Link href="#overiew" target="_self">Overview</Link>}
+- {<Link href="#component-props" target="_self">Component Props</Link>}
+- {<Link href="#accessibility" target="_self">Accessibility</Link>}
+- {<Link href="#variants" target="_self">Variants</Link>}
+- {<Link href="#banner-heading" target="_self">Banner Heading</Link>}
+- {<Link href="#banner-icon" target="_self">Banner Icon</Link>}
+- {<Link href="#with-html-content" target="_self">With HTML Content</Link>}
+- {<Link href="#custom-color" target="_self">Custom Color</Link>}
+- {<Link href="#dismissible" target="_self">Dismissible</Link>}
+- {<Link href="#changelog" target="_self">Changelog</Link>}
+
+## Overview
+
+<Description of={BannerStories} />
+
+**IMPORTANT:** The `Banner` component mirrors the `Notification` component but
+the `Notification` component should be used for more global messaging at the top
+of the screen. The `Banner` component should be used for all messaging that
+sits within the flow of the page.
+
+## Component Props
+
+<Canvas of={BannerStories.WithControls} />
+
+<ArgTypes of={BannerStories.WithControls} />
+
+## Accessibility
+
+The `Banner` component renders with an HTML `aside` element as its
+wrapper. This is an HTML landmark element that is similar to adding an attribute
+of `role="complementary"`. For accessibility purposes, landmark elements should
+not be rendered inside other landmark elements such as the `header` and `footer`
+landmark elements. Adding a `Banner` component inside an HTML `main`
+landmark element is acceptable.
+
+### ARIA Label
```

Comment:
If this is required, should it have a red asterisk next to it in the props table?

Reply:
Updated the language to better be reflect that it's optional.